### ISSUE TITLE
Function _convert_node_attr_types(G, dtypes=None) in io.py doesn't work with boolean attributes

### ISSUE THREAD

**Issue by guibar** [gboeing/osmnx#665]
If I add a boolean field on nodes and save it to graphml, I don't get it back when I read from the file with:

`osmnx.io.load_graphml(graphml_path, node_dtypes=node_dtypes)`

even though I have added by field in node_dtypes with the bool type. 

I can see that my fields are saved correctly as 'True' or 'False' in the graphml file. 

But when reading from the file, it fails because the following code in _convert_node_attr_types:

```
    for _, data in G.nodes(data=True):
        for attr in dtypes:
            if attr in data:
                data[attr] = dtypes[attr](data[attr])
```
doesn't work with bool i.e. bool('False') doesn't return False but True as can be seen here:

https://stackoverflow.com/questions/715417/converting-from-a-string-to-boolean-in-python

So all my field are set to True. 

We can discuss a way to fix this if you want. Although they are going to be hacky given that the really good solution would be for bool('False') to be False.

**Comment by gboeing** [comment]
Interesting. Thanks for flagging. Do you have some ideas for a fix? Currently, OSMnx manually converts a couple of bool types via literal_eval.

**Comment by guibar** [comment]
I think you are in a better position than me to decide what is the least
hacky way. Can you point me to some use of literal_eval?

**Comment by gboeing** [comment]
See [here](https://github.com/gboeing/osmnx/blob/v1.0.1/osmnx/io.py#L228-L233) and [here](https://github.com/gboeing/osmnx/blob/v1.0.1/osmnx/io.py#L304-L310) for two instances where we `literal_eval` to convert "True"/"False" strings -> bool.

I think some kind of generalized solution like this would be the best bet. Like, if the user specified `bool` as the dtype value, then `literal_eval` it instead of using `bool` as the converter. Something like that? This solution does assume that the boolean values in the strings are "True" or "False", and nothing else.

**Comment by gboeing** [comment]
Here's a quick stab at a minimal reproducible test we'd need to get working:

```python
import networkx as nx
import numpy as np
import osmnx as ox

fp = './graph.graphml'
attr_name = 'test_bool'
G = ox.graph_from_place('Piedmont, CA, USA', network_type='drive')

# create random boolean node/edge attributes
bools = np.random.randint(0, 2, len(G.nodes))
node_attrs = {n:bool(b) for n, b in zip(G.nodes, bools)}
nx.set_node_attributes(G, node_attrs, attr_name)
bools = np.random.randint(0, 2, len(G.edges))
edge_attrs = {n:bool(b) for n, b in zip(G.edges, bools)}
nx.set_edge_attributes(G, edge_attrs, attr_name)

# save graph to disk then reload
ox.save_graphml(G, filepath=fp)
G2 = ox.load_graphml(fp,
                     node_dtypes={attr_name:bool},
                     edge_dtypes={attr_name:bool})

# ensure all the values are of type bool in both graphs nodes and edges
assert all(isinstance(value, bool) for value in nx.get_node_attributes(G, attr_name).values())
assert all(isinstance(value, bool) for value in nx.get_edge_attributes(G, attr_name).values())
assert all(isinstance(value, bool) for value in nx.get_node_attributes(G2, attr_name).values())
assert all(isinstance(value, bool) for value in nx.get_edge_attributes(G2, attr_name).values())

# ensure all node/edge values are equal between the two graphs
for k, v in nx.get_node_attributes(G, attr_name).items():
    assert v == G2.nodes[k][attr_name]
for k, v in nx.get_edge_attributes(G, attr_name).items():
    assert v == G2.edges[k][attr_name]
```

**Comment by gboeing** [comment]
@guibar note that you can currently get this working by passing literal_eval as your data type for boolean attributes:

```python
import ast
G2 = ox.load_graphml(fp,
                     node_dtypes={attr_name:ast.literal_eval},
                     edge_dtypes={attr_name:ast.literal_eval})
```

So one option could be: change nothing in the codebase but document that users should pass literal_eval instead of bool for correct boolean value conversion.

**Comment by gboeing** [comment]
See proposed fix in #670

**Comment by gboeing** [comment]
See amended fix in #672.
